examples/streams/streams.bal

```diff
@@ -0,0 +1,120 @@
+import ballerina/io;
+
+//Defines an object called OddNumberGenerator. Each object has its own next() method,
+//which gets invoked when the stream's next() gets called.
+type OddNumberGenerator object {
+    int i = 1;
+    public function next() returns record {|int value;|}|error? {
+        self.i += 2;
+        return {value: self.i};
+    }
+};
+
+type ResultValue record {|
+    int value;
+|};
+
+type Student record {
+    string firstName;
+    string lastName;
+    float score;
+};
+
+type StudentValue record {|
+    Student value;
+|};
+
+type FullName record {|
+    string firstName;
+    string lastName;
+|};
+
+type Subscription record {|
+    string firstName;
+    string lastName;
+    float score;
+    string degree;
+|};
+
+public function main() {
+    io:println("Creating a stream using stream constructor: ");
```

Comment:
Let's remove output if unnecessary and make them comments maybe? That way they'll get rendered side-by-side.